provider/openstack/firewaller.go

```diff
@@ -748,17 +774,32 @@ func (c *neutronFirewaller) updateGroupControllerUUID(group *neutron.SecurityGro
 
 // OpenPorts implements Firewaller interface.
 func (c *neutronFirewaller) OpenPorts(ctx context.ProviderCallContext, rules []network.IngressRule) error {
-	return c.openPorts(c.openPortsInGroup, rules)
+	err := c.openPorts(ctx, c.openPortsInGroup, rules)
+	if err != nil {
+		common.HandleCredentialError(IsAuthorisationFailure, err, ctx)
+		return errors.Trace(err)
+	}
+	return nil
 }
 
 // ClosePorts implements Firewaller interface.
 func (c *neutronFirewaller) ClosePorts(ctx context.ProviderCallContext, rules []network.IngressRule) error {
-	return c.closePorts(c.closePortsInGroup, rules)
+	err := c.closePorts(ctx, c.closePortsInGroup, rules)
+	if err != nil {
+		common.HandleCredentialError(IsAuthorisationFailure, err, ctx)
+		return errors.Trace(err)
+	}
+	return nil
 }
 
 // IngressRules implements Firewaller interface.
 func (c *neutronFirewaller) IngressRules(ctx context.ProviderCallContext) ([]network.IngressRule, error) {
-	return c.ingressRules(c.ingressRulesInGroup)
+	rules, err := c.ingressRules(ctx, c.ingressRulesInGroup)
+	if err != nil {
+		common.HandleCredentialError(IsAuthorisationFailure, err, ctx)
+		return rules, errors.Trace(err)
```

Comment:
This line can just be moved below replacing
return rules, nil
(here and elsewhere)
since errors.Trace(nil) returns nil

Reply:
True. However, isn't it clearer to return nil when you know that there is no errors?